pkg/storage/juicefs.go

```diff
@@ -122,13 +123,29 @@ func (s *JuiceFsStorage) Format(fsName string) error {
 }
 
 func (s *JuiceFsStorage) Unmount(localPath string) error {
-	cmd := exec.Command("juicefs", "umount", "--force", localPath)
+	juiceFsUmount := func() error {
+		cmd := exec.Command("juicefs", "umount", localPath)
 
-	output, err := cmd.CombinedOutput()
+		output, err := cmd.CombinedOutput()
+		if err != nil {
+			log.Printf("error executing juicefs umount: %v, output: %s", err, string(output))
+			return err
+		}
+
+		log.Printf("JuiceFS filesystem unmounted from: '%s'\n", localPath)
+		return nil
+	}
+
+	err := backoff.Retry(juiceFsUmount, backoff.WithMaxRetries(backoff.NewConstantBackOff(1*time.Second), 10))
+	if err == nil {
+		return nil
+	}
+
+	// Forcefully kill the fuse mount devices
```

Comment:
Could this affect other fuse mounts of other workers running on multi-worker nodes?

Reply:
It only kills pods that it finds using that file with fuser /dev/fuse which results to only files on that pod